cdap-ui/app/cdap/components/AbstractWidget/KeyValueWidget/index.tsx

```diff
@@ -76,3 +78,13 @@ export default function KeyValueWidget(props) {
   ...WIDGET_PROPTYPES,
   isEncoded: PropTypes.bool,
 };
+
+(KeyValueWidget as any).getWidgetAttributes = () => {
+  return {
+    'key-placeholder': { type: 'string', required: false },
+    'value-placeholder': { type: 'string', required: false },
+    'kv-delimiter': { type: 'string', required: false },
+    delimiter: { type: 'string', required: false },
+    showDelimiter: { type: 'boolean', required: false },
```

Comment:
where do we use this?

Reply:
I've included this because I've noticed that this is occasionally included in JSON configuration.

https://github.com/cdapio/hydrator-plugins/blob/544519f394360fb7a045e6be02fe0afea4f40c27/core-plugins/widgets/Cube-batchsink.json#L32